### DISCUSSION TITLE
Snowflake support in sequelize - limitations and gaps

### DISCUSSION THREAD

**Discussion by sfc-gh-dgadomski** [sequelize/sequelize#15099] [Help & Questions] [answered]
Hello, 

I'm using sequelize for my project and it's great. Thank you for maintaining it :) I see that the support is experimental and limited for now. My question here is about those limitation as I would like to contribute in this area. What do you think would bring the most value? Are you aware of some important gaps?

For now I've already fixed a recent bug (https://github.com/sequelize/sequelize/pull/14930 & https://github.com/sequelize/sequelize/pull/15087) and I also see some unimplemented methods in the dialect, like:

```
createSchema() {
  return 'SHOW TABLES';
}

showSchemasQuery() {
  return 'SHOW TABLES'; }
```

Tagging @jesse23 as git blame in snowflake files points to you :) Maybe you could help determine the possible work to do.

Thanks!

**Comment by WikiRik** [comment, marked as answer]
I think the main limitation is that we (as maintainers) do not have access to a snowflake instance to run integration tests on and therefore support might break without us knowing. We are relying on the community to notice this and help with fixing these issues.

For example we know that snowflake support will break in the v7-alpha after merging #14505 but we don't exactly know in what way.

What would help us the most to improve the support therefore is access to an instance for running our integration tests through the GitHub Actions CI.

If you want to work on code changes to benefit the support I think you can look at the tests that are being skipped/altered for snowflake and see if that's due to snowflake itself or the implementation in sequelize. Other points might have been mentioned in the PR that initially added the support; https://github.com/sequelize/sequelize/pull/13406

**Reply by sdepold** [reply]
I can only really underline what Rik is saying. It is close to impossible to properly verify that existing functionality is still working and that new features are behaving as expected without access to a proper testing environment.

So if you happen to have a contact at Snowflake or want to start the conversation with them to get permanent access to a database instance, this would massively help us :)

**Reply by jesse23** [reply]
@sfc-gh-dgadomski Hey I tried last year and early this year and not able to get a sandbox....Yes if we can get a sandbox support that will be awesome!

I don't have access too at this moment since I don't work on Dynatrace anymore...
